### PR TITLE
scenario_description_added

### PR DESCRIPTION
**Pull request by ashish-1600**
### Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Description

Please include a summary of the change and which issue it addresses. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

### Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/uptrain-ai/uptrain/blob/main/CONTRIBUTING.md) document.
- [ ] My code follows the code style (BLACK) of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the documentation accordingly.
- [ ] I have added an appropriate CHANGELOG entry.

### Author's Note

Please provide any additional information, questions, or concerns you have about this pull request.

### CODE REVIEW

File: uptrain/framework/evalllm.py [hunk 1]
```diff
@@ -79,9 +80,9 @@ def evaluate(
                 req_attrs.update([schema.conversation])
 
             if isinstance(m, ParametricEval):
-                ser_checks.append({"check_name": m.__class__.__name__, **m.dict()})
+                ser_checks.append({"check_name": m.__class__.__name__, **m.dict(), "scenario_description": scenario_description})
```
Comment (sourabhagr) [inline uptrain/framework/evalllm.py:83]: Why not add scenario_description to m.dict()? @ashish-1600

File: uptrain/framework/evalllm.py [hunk 2]
```diff
@@ -79,9 +80,9 @@ def evaluate(
...
             elif isinstance(m, Evals):
-                ser_checks.append(m.value)
+                ser_checks.append({"check_name": m.value, "scenario_description": scenario_description})
```
Comment (sourabhagr) [inline uptrain/framework/evalllm.py:85]: Why not: 

**{"scenario_description": scenario_description}

File: uptrain/framework/remote.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def __init__(self, settings: Settings = None) -> None:
     def evaluate(
         self,
         data: list[dict],
-        checks: list[t.Union[Evals, ParametricEval]],
+        checks: list[t.Union[Evals, ParametricEval, dict]],
```
Comment (sourabhagr) [inline uptrain/framework/remote.py:55]: Should take scenario_description as input?

File: uptrain/framework/evalllm.py (hunk 2 above)
Comment (sourabhagr) [reply uptrain/framework/evalllm.py:85]: elif isinstance(m, Evals):
         dictn = {"scenario_description": scenario_description}
         ser_checks.append({"check_name": m.value, **dictn)

File: uptrain/framework/evalllm.py (hunk 1 above)
Comment (sourabhagr) [reply uptrain/framework/evalllm.py:83]: if isinstance(m, ParametricEval):
         dictn = m.dict()
         dictn.update({"scenario_description": scenario_description})
         ser_checks.append({"check_name": m.__class__.__name__, **dictn)

Comment (sourabhagr) [reply uptrain/framework/evalllm.py:83]: @ashish-1600

File: uptrain/framework/remote.py (hunk 3 above)
Comment (ashish-1600) [reply uptrain/framework/remote.py:55]: The scenario_description is already present in checks. It is populated in EvalLLM.

**Review by sourabhagr — APPROVED** [review]

**Review by sourabhagr — APPROVED** [review]

**Pull request merged**